### PR TITLE
Added feature in weather-mv to extract specific date's data from zarr files.

### PR DESCRIPTION
**Pull request by dabhicusp**
I have added a feature in `weather-mv` which can allows users to access data for a specific date from Zarr files. In the existing codebase user have no choice for working with some chunks of the `zarr` file so user have to work with whole zarr file every time. 
One use-case: If user want to ingest specific date's data from the zarr file to BQ then he/she can do this using of this code.

### CODE REVIEW

File: weather_mv/loader_pipeline/sinks.py [hunk 1]
```diff
@@ -406,7 +409,19 @@ def open_dataset(uri: str,
     """Open the dataset at 'uri' and return a xarray.Dataset."""
     try:
         if is_zarr:
+            start_date = end_date = None
+            if 'start_date' in open_dataset_kwargs:
+                start_date = open_dataset_kwargs['start_date']
+                if open_dataset_kwargs is not None:
+                    del open_dataset_kwargs['start_date']
```
Comment (alxmrs) [inline weather_mv/loader_pipeline/sinks.py:416]: Can you call something like `pop()` here?

File: weather_mv/loader_pipeline/sinks.py [hunk 2]
```diff
@@ -406,7 +409,19 @@ def open_dataset(uri: str,
...
+            if 'end_date' in open_dataset_kwargs:
+                end_date = open_dataset_kwargs['end_date']
+                if open_dataset_kwargs is not None:
+                    del open_dataset_kwargs['end_date']
             ds: xr.Dataset = _add_is_normalized_attr(xr.open_dataset(uri, engine='zarr', **open_dataset_kwargs), False)
+            if start_date is not None and end_date is not None:
+                timedelta = pd.date_range(start_date, end_date, periods=25, inclusive='left')
+                ds = ds.sel(time=timedelta)
```
Comment (alxmrs) [inline weather_mv/loader_pipeline/sinks.py:424]: TIL! this is slick!

File: weather_mv/loader_pipeline/sinks.py [hunk 3]
```diff
@@ -406,7 +409,19 @@ def open_dataset(uri: str,
...
+                start_date = open_dataset_kwargs['start_date']
+                if open_dataset_kwargs is not None:
+                    del open_dataset_kwargs['start_date']
+            if 'end_date' in open_dataset_kwargs:
+                end_date = open_dataset_kwargs['end_date']
+                if open_dataset_kwargs is not None:
+                    del open_dataset_kwargs['end_date']
             ds: xr.Dataset = _add_is_normalized_attr(xr.open_dataset(uri, engine='zarr', **open_dataset_kwargs), False)
+            if start_date is not None and end_date is not None:
+                timedelta = pd.date_range(start_date, end_date, periods=25, inclusive='left')
```
Comment (alxmrs) [inline weather_mv/loader_pipeline/sinks.py:423]: nit: let's call this `time_range` or similar.

File: weather_mv/loader_pipeline/sinks.py (hunk 2 above)
Comment (dabhicusp) [reply weather_mv/loader_pipeline/sinks.py:424]: It's my pleasure. 😄

Comment (alxmrs) [reply weather_mv/loader_pipeline/sinks.py:424]: FWIW: I think this also may be overkill. IIUC, you could pas the `start_date` and `end_date` from the `pd.date_range` directly into the `sel`, e.g.: 

```
ds = ds.sel(time=slice(start_date, end_date))
```

File: weather_mv/loader_pipeline/sinks.py (hunk 1 above)
Comment (dabhicusp) [reply weather_mv/loader_pipeline/sinks.py]: Thank you for suggestions.

File: weather_mv/loader_pipeline/sinks.py [hunk 4]
```diff
@@ -406,7 +409,19 @@ def open_dataset(uri: str,
     """Open the dataset at 'uri' and return a xarray.Dataset."""
     try:
         if is_zarr:
+            start_date = end_date = None
+            if 'start_date' in open_dataset_kwargs:
+                start_date = open_dataset_kwargs['start_date']
+                if open_dataset_kwargs is not None:
+                    open_dataset_kwargs.pop('start_date', None)
```
Comment (alxmrs) [inline weather_mv/loader_pipeline/sinks.py:416]:
```suggestion
                start_date = open_dataset_kwargs.pop('start_date', None)                    
```

File: weather_mv/loader_pipeline/sinks.py [hunk 5]
```diff
@@ -406,7 +409,19 @@ def open_dataset(uri: str,
...
+            if 'end_date' in open_dataset_kwargs:
+                end_date = open_dataset_kwargs['end_date']
+                if open_dataset_kwargs is not None:
+                    open_dataset_kwargs.pop('end_date', None)
             ds: xr.Dataset = _add_is_normalized_attr(xr.open_dataset(uri, engine='zarr', **open_dataset_kwargs), False)
+            if start_date is not None and end_date is not None:
+                time_range = pd.date_range(start_date, end_date, periods=25, inclusive='left')
+                ds = ds.sel(time=time_range)
```
Comment (alxmrs) [inline weather_mv/loader_pipeline/sinks.py:424]: Did you see my other comment? why not just pass `start_date` and `end_date` into the `sel` via slice? 
```suggestion
                ds = ds.sel(time=slice(start_date, end_date))
```

File: weather_mv/loader_pipeline/sinks.py (hunk 4 above)
Comment (dabhicusp) [reply weather_mv/loader_pipeline/sinks.py:416]: Hi @alxmrs, it's important to ensure that a dictionary is not **empty** before using the `pop` operation directly on it. Therefore, we have to must add a condition like `if open_dataset_kwargs is not None:` before using `pop` operation.

Despite making code updates, I aimed to keep the code as minimal as possible.

File: weather_mv/loader_pipeline/sinks.py (hunk 5 above)
Comment (dabhicusp) [reply weather_mv/loader_pipeline/sinks.py]: Sorry I forgot the above comments. now it's changed.

**Review by mahrsee1997 — APPROVED** [review]
LGTM.

**Review by alxmrs — APPROVED** [review]
Perfect, change looks great!

**Pull request merged**